### PR TITLE
Implement Proposal execution

### LINKED ISSUE #38: Implement MsgExec functionality
* MsgExec
As a user on the chain I want to be able to execute a proposal before the timeout when it was accepted so that the group's decision becomes
reality.

This includes implementing payload message execution (routing, auth)

needed-by #31

### PR DESCRIPTION
**Pull request by alpe**
After #66

Resolves #38

### CODE REVIEW

File: incubator/group/proposal_executor_test.go [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+	}{
+		"all good": {
+			srcAccount: []byte("my-group-acct-addrss"),
+			srcMsgs:    []sdk.Msg{MyMsg{[]sdk.AccAddress{[]byte("my-group-acct-addrss")}}},
+			srcHandler: mockHandler(&sdk.Result{}, nil),
+		},
+		"not authz by group account": {
+			srcAccount: []byte("my-group-acct-addrss"),
+			srcMsgs:    []sdk.Msg{MyMsg{[]sdk.AccAddress{[]byte("any--other---address")}}},
+			srcHandler: alwaysPanicHandler(),
```
Comment (aaronc) [inline incubator/group/proposal_executor_test.go:26]: shouldn't this be `mockHandler` if we actually want to test this case? (panic might get caught by the router)

**Review by aaronc — COMMENTED** [review]
Mostly looks good.

I see in the test cases that there are a number of edge cases around how proposal execution behaves. Are these listed out and documented somewhere?

Also I'm having trouble understanding where all the `testdata` stuff is used in this PR.

File: incubator/group/proposal_executor_test.go (hunk 1 above)
Comment (alpe) [reply incubator/group/proposal_executor_test.go:26]: In this scenario the authentication decorator is tested. The handler should not be called as the payload message "signer" is not the groupAccount. This prevents access to random accounts.
The `alwaysPanicHandler` makes sense as it would raise a panic and not an error. The panic would not be handled by the router.

**Comment by alpe** [comment]
As with the other PRs there is no "proper documentation" written. I have tried to define test cases for all the scenarios I could imagine so that they are a specification as code. Probably not what you were looking for. 
I think we should make documentation a first class task. But before, I would appreciate if we can be more precise what kind of documentation, where we host it and who we are targeting as readers. Otherwise it would become an epic. We currently have only https://github.com/regen-network/cosmos-modules/issues/40 which is not on the critical path.

The `testdata` messages are used for the tests in ProposalExecutor and Keeper. I see them also as extension point so that people may use them for their test scenarios.

**Comment by aaronc** [comment]
@clevinson can you comment on documentation here?

**Review by aaronc — APPROVED** [review]
Approving this, but would be great if we can track all the edge cases covered here as a documentation TODO. @clevinson maybe you can flag this in an issue and link to the test cases covered here?

**Comment by alpe** [comment]
I have created an issue and linked to the comment
